### PR TITLE
timew-sync-client: init at 1.0.1-unstable-2024-08-05

### PR DESCRIPTION
**Pull request by evris99**
## Description of changes

This package is the client component of timewarrior sync.
The code can be found [here](https://github.com/timewarrior-synchronize/timew-sync-client).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/timew-sync-client/default.nix
Comment (Aleksanaa) [inline pkgs/applications/misc/timew-sync-client/default.nix:8]:
```suggestion
  version = "1.0.1-unstable-2024-06-19";
```

File: pkgs/applications/misc/timew-sync-client/default.nix
Comment (Aleksanaa) [inline pkgs/applications/misc/timew-sync-client/default.nix:34]: Better to avoid `with lib`

File: pkgs/applications/misc/timew-sync-client/default.nix
Comment (evris99) [reply pkgs/applications/misc/timew-sync-client/default.nix]: Done

File: pkgs/applications/misc/timew-sync-client/default.nix
Comment (evris99) [reply pkgs/applications/misc/timew-sync-client/default.nix]: Done

**Comment by Aleksanaa** [comment]
@ofborg build timew-sync-client

**Comment by evris99** [comment]
Is there something else I should change from my side or should I just wait for the merge?

File: pkgs/applications/misc/timew-sync-client/default.nix
Comment (errnoh) [inline pkgs/applications/misc/timew-sync-client/default.nix:14]:
```suggestion
  version = "1.0.1-unstable-2024-08-05";

  src = fetchFromGitHub {
    owner = "timewarrior-synchronize";
    repo = "timew-sync-client";
    rev = "de3442bd29b071f54cd1e10af99f3378a83b4794";
    hash = "sha256-AKRAMEUTIPvR+kaEZZYjd4II2KzYZTwRgGzFMGD5aio=";
```

Newer version as there have been multiple commits since the creation of this PR

File: pkgs/applications/misc/timew-sync-client/default.nix
Comment (errnoh) [inline pkgs/applications/misc/timew-sync-client/default.nix:27]:
```suggestion
```

I don't think this is necessary, at least haven't been listing it as dependency on my local version which has been working fine.

File: pkgs/applications/misc/timew-sync-client/default.nix
Comment (errnoh) [inline pkgs/applications/misc/timew-sync-client/default.nix:32]:
```suggestion
```

Doesn't seem necessary to disable tests when there aren't any

File: pkgs/applications/misc/timew-sync-client/default.nix
Comment (errnoh) [inline pkgs/applications/misc/timew-sync-client/default.nix:39]:
```suggestion
    maintainers = with lib.maintainers; [ evris99 errnoh ];
```

I don't mind co-maintaining as I'm actively using almost identical package already.

**Review by errnoh — CHANGES_REQUESTED** [review]
Merging the PR seems to have taken a long time, added suggestions for changes to incorporate commits since the creation of the PR.

Btw the title probably should reflect the actual `version` string?

**Comment by errnoh** [comment]
And quite often unless there's another interested party with merge rights, you'll get PRs merged faster by advertising the PR on threads like [this](https://discourse.nixos.org/t/prs-ready-for-review/3032/4650) :)

File: pkgs/applications/misc/timew-sync-client/default.nix
Comment (FliegendeWurst) [reply pkgs/applications/misc/timew-sync-client/default.nix:32]: It is good to document why checks are disabled.

**Comment by evris99** [comment]
Thank you @errnoh for your helpful suggestions!

I incorporated them into the PR.

**Comment by errnoh** [comment]
Nice, we're almost there!

Couple remaining things: 
- since the creation of this commit nixpkgs has started to only use `pkgs/by-name` so you'll need to move the package from `pkgs/applications/misc/timew-sync-client` to `pkgs/by-name/ti/timew-sync-client`.
- the main nix file should be `package.nix` instead of `default.nix`.

After those changes are done you should squash your commits, be sure to have the correct version in the commit line, and change the title of this PR to match as well.

**Comment by evris99** [comment]
Ok I squashed the commit and updated the path.

**Review by errnoh — APPROVED** [review]
Confirmed to work on x86_64-linux

**Review by errnoh — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**
